src/main/state/nodes.ts

```diff
@@ -197,3 +199,31 @@ export const removeNode = (nodeId: NodeId) => {
   store.set(USER_NODES_KEY, { nodes, nodeIds: newNodeIds });
   return nodeToRemove;
 };
+
+powerMonitor.on('shutdown', () => {
+  restartNodes('shutdown');
```

Comment:
actually, we would want to shutdown nodes here and on suspend and `stoppedBy` should be set to 'shutdown'